### ISSUE TITLE
Silent playback in very rare situations 

### ISSUE THREAD

**Issue by lstrsrt** [fooyin/fooyin#380]
Songs are rarely -- I'd say maybe 1 out of a 100 times -- played back with no sound. It can happen when clicking on a song or just advancing in a playlist. Simply restarting the song or skipping to a different point fixes it. Because this bug is so uncommon, I haven't been able to properly debug or find preconditions to trigger it, but the fact that it's so rare makes me think it might be a race condition.

 - Setup: Fedora 40 (GNOME, Pipewire)
 - Version: `17d6669277c7606b7fe2229ffffed5f57d6311fd`, but also observed on earlier commits.

**Comment by ludouzi** [comment]
That definitely sounds like a race condition. 

If silent playback is the only issue, and the playback position is still being updated, then that may suggest an issue in AudioRenderer somewhere, or one of it's methods isn't being called in time.

I've added a few log messages in https://github.com/fooyin/fooyin/commit/8600ca86df416865b523f44443fb19fcbbca9554 to try and catch some obvious errors in the write loop. I have a few ideas for attempting to reproduce this which I'll look into later.

**Comment by lstrsrt** [comment]
Unrelated, but another issue I encountered today: fooyin doesn't build anymore after updating to cmake 3.30.

```
FAILED: src/utils/fooyin_utils_autogen/timestamp src/utils/fooyin_utils_autogen/mocs_compilation.cpp /home/user/src/fooyin/build-dev/src/utils/fooyin_utils_autogen/timestamp /home/user/src/fooyin/build-dev/src/utils/fooyin_utils_autogen/mocs_compilation.cpp 
cd /home/user/src/fooyin/build-dev/src/utils && /usr/bin/cmake -E cmake_autogen /home/user/src/fooyin/build-dev/src/utils/CMakeFiles/fooyin_utils_autogen.dir/AutogenInfo.json Debug && /usr/bin/cmake -E touch /home/user/src/fooyin/build-dev/src/utils/fooyin_utils_autogen/timestamp && /usr/bin/cmake -E cmake_transform_depfile Ninja gccdepfile /home/user/src/fooyin /home/user/src/fooyin/src/utils /home/user/src/fooyin/build-dev /home/user/src/fooyin/build-dev/src/utils /home/user/src/fooyin/build-dev/src/utils/fooyin_utils_autogen/deps /home/user/src/fooyin/build-dev/CMakeFiles/d/8f1e6cf8a1f3a8ab7da1af1c76fada6accd7f1e61b044877b3f3fe1842fd5e71.d

AutoGen error
-------------
Info error in info file
"/home/user/src/fooyin/build-dev/src/utils/CMakeFiles/fooyin_utils_autogen.dir/AutogenInfo.json":
CROSS_CONFIG is not a boolean.
```

It builds again after downgrading to 3.28, so there was probably some breaking change in the new cmake release (possibly related to compatibility with MOC).

**Comment by ludouzi** [comment]
Have you tried clearing the CMake configuration or a fresh build? I've been using CMake 3.30 since early October and haven't encountered such errors.

**Comment by lstrsrt** [comment]
Ok, fully deleting and regenerating the build directory worked. I only tried `--target clean` before.

**Comment by ludouzi** [comment]
That means AudioRenderer is not the source of the issue. The renderer is correctly started but buffers aren't being queued to be played.

**Comment by lstrsrt** [comment]
Yeah I was able to replicate it by commenting out the call to readNextBuffer in the playback engine.

**Comment by x80486** [comment]
I experience this quite often — five to ten times a day. I usually don't notice it because I leave the music playing in the background, and I only realize something is wrong when everything goes dead silent.

I'm using the Flatpak version of the application, and I only play FLAC file — not sure if that's relevant.

**Comment by ludouzi** [comment]
I've been able to reproduce this much more consistently since moving my library to a NAS. It's also more likely to happen when playing tracks in archives. There's definitely a race condition somewhere.

**Comment by ludouzi** [comment]
May have finally found the cause - unsigned integer underflow.

I can't reproduce this anymore, so here's hoping that solves it. We can leave this issue open for a while just in case.

**Comment by lstrsrt** [comment]
Seems like a good fix for sure. I have not had this happen since applying the patch last week, so I'll close this.
